core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -49,7 +49,15 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
           .getInputMetricsForReadMethod(inputMetrics.readMethod)
         existingMetrics.incBytesRead(inputMetrics.bytesRead)
 
-        val iter = blockResult.data.asInstanceOf[Iterator[T]]
+        val buf = blockResult.data.toArray
+        if (rdd.sparkContext.getConf.getBoolean("spark.rdd.remoteblock.cache", false)) {
```

Comment:
Does this need a conf option? is there a drawback to doing this all the time?
You could collapse these into one conditional too, but that's trivial